docroot/modules/custom/dct_sponsors/dct_sponsors.module

```diff
@@ -56,6 +56,8 @@ function dct_sponsors_form_alter(&$form, FormStateInterface $form_state, $form_i
       unset($form['revision_information']);
       unset($form['revision']);
       unset($form['field_type']);
+      unset($form['field_sponsor_editor']);
+      unset($form['actions']['delete']);
 
```

Comment:
The permission has to be modified too, like that the user if knows the link, can delete the node.